app/src/main/java/com/example/mplayer/play/PlayActivity.java

```diff
@@ -0,0 +1,177 @@
+import static com.example.mplayer.utils.Constant.Bundle.BUNDLE_LIST_MUSIC;
+
+public class PlayActivity extends AppCompatActivity implements View.OnClickListener, PlayContract.View,
+        ServiceConnection, SeekBar.OnSeekBarChangeListener {
+    private MusicService mMusicService;
+    private ImageButton mButtonPrevious;
+    private ImageButton mButtonPlay;
+    private ImageButton mButtonNext;
+    private SeekBar mSeekBar;
+    private TextView mTextStartTime;
+    private TextView mTextTotalTime;
+    private TextView mTextSongName;
+    private PlayContract.Presenter mPresenter;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        createService();
+        registerReceiver(receiver, new IntentFilter(ACTION_NOTIFICATION_BUTTON_CLICK));
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_play);
+        getControls();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+    }
+
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.button_previous:
+                mPresenter.previous();
+                break;
+
+            case R.id.button_play:
+                mPresenter.play();
+                mMusicService.showNotification();
+                break;
+
+            case R.id.button_next:
+                mPresenter.next();
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    public void createService() {
+        Intent musicSerivce = new Intent(this, MusicService.class);
+        startService(musicSerivce);
+        bindService(musicSerivce, this, Context.BIND_AUTO_CREATE);
+    }
+
+    @Override
+    protected void onStop() {
+        unbindService(this);
+        unregisterReceiver(receiver);
+        super.onStop();
+    }
+
+    @Override
+    public void onServiceConnected(ComponentName name, IBinder service) {
+        mMusicService = ((MusicService.LocalBinder) service).getService();
+        mPresenter = new PlayPresenter(SongRepository.getInstance(SongLocalDataSource.getInstance(this)),
+                this, mMusicService.getMusicController());
+        mMusicService.getMusicController().setData(getIntent().<Song>getParcelableArrayListExtra(BUNDLE_LIST_MUSIC),
+                getIntent().getIntExtra(BUNDLE_CURRENT_POSITION, 0));
+        updateUi();
+        mMusicService.showNotification();
+    }
+
+    @Override
+    public void onServiceDisconnected(ComponentName name) {
+
+    }
+
+    @Override
+    public void updateUi() {
+        if (mMusicService.getMusicController().isPlaying()) {
+            mButtonPlay.setImageResource(R.drawable.icon_pause);
+        } else {
+            mButtonPlay.setImageResource(R.drawable.icon_play);
+        }
+        final Handler mHandler = new Handler();
+        mHandler.postDelayed(new Runnable() {
```

Comment:
sao ko sử dụng method post mà lại postDelay(0

Reply:
Không delay thì có gây nặng cho main-thread không anh ?

Reply:
@DoanVanToan

Reply:
postDelay ma truyen vao thời gian delay là 0 thi sao k dùng post luôn. Ý a Toản là như vậy

Reply:
@manhpv01 delay 0 chỉ là lần đầu chạy để update nhanh hơn.

Reply:
vậy bạn viết như này cũng được 
 handle.post(new Runable{
     run(){
   handle.postDelay(this, 1000)
}
})

Reply:
@manhpv01 à @@ Thanks b